app/src/main/java/fr/free/nrw/commons/theme/NavigationBaseActivity.java

```diff
@@ -68,12 +70,20 @@ private void setUserName() {
 
         View navHeaderView = navigationView.getHeaderView(0);
         TextView username = navHeaderView.findViewById(R.id.username);
-
         AccountManager accountManager = AccountManager.get(this);
         Account[] allAccounts = accountManager.getAccountsByType(AccountUtil.ACCOUNT_TYPE);
         if (allAccounts.length != 0) {
             username.setText(allAccounts[0].name);
         }
+        ImageView userIcon = navHeaderView.findViewById(R.id.user_icon);
+        userIcon.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                drawerLayout.closeDrawer(navigationView);
+                startActivityWithFlags(NavigationBaseActivity.this, AchievementsActivity.class, Intent.FLAG_ACTIVITY_CLEAR_TOP,
```

Comment:
IMO, just the activity itself ie `AchievementsActivity` in this case should know how it can be started. It could simply expose a static method something like: 

```
AchievementsActivity.startYourself(context);
```

Few of the newer activities in the app already use this pattern. You can take a look at them. Its merely a convention but I find it very useful for activities that can be opened in multiple ways(ie by accepting different init params).